Hi there!

This is my exam solution of "Module 5 - Next Generation sequencing" from the CQ-Course "ABI-2025-2 Applied bioinformatics and biostatistics".

It is a nextflow pipeline which takes some sequences from a colleague (from url) and compares them (alignment via mafft: https://mafft.cbrc.jp/alignment/software/) to the reference hepatitis genome ("M21012") from the NCBI databank. If you want to use local sequences on your harddrive for comparison, put them here: "${projectDir}/input". Afterwards the alignment gets cleaned via trimal (https://trimal.cgenomics.org/). The output-files are just a cleaned-up fasta file and a html report. However, intermediate outputs of each process can be generated by removing the comments in front of //publishDir.

You can choose different reference genomes by adding --accession "reference number" in your command line. All tools you need are stored in containers. To use them, you need singularity. The setup for singularity is done in the nextflow.config file (profile: singularity) If you want to run the nextflow file, use the following command:

nextflow run -profile singularity exam_Werny.nf //for the standard reference genome "M21012")

nextflow run -profile singularity exam_Werny.nf --accession "reference number" (for a comparison to a reference number of your own choice)

