Hi there!

This is my exam solution of "Module 5 - Next Generation sequencing" from the CQ-Course "ABI-2025-2 Applied bioinformatics and biostatistics".

It is a nextflow pipeline which takes some sequences from a collegue and compares them (alignment via mafft: https://mafft.cbrc.jp/alignment/software/) to the reference hepatitis genome ("M21012") from the NCBI databank. 
Afterwards the alignment gets cleanded via trimal (https://trimal.cgenomics.org/). 
The output-files are just a cleaned-up fasta file and a html report. However, intermediate outputs of each process can be generated by removing the comments infront of //publishDir.

You can choose different reference genomes by adding --accession "reference number" in your commandline. 
All tools you need are stored in containers. To use them, you need singularity. 
